flow/connectors/bigquery/bigquery.go

```diff
@@ -141,6 +141,51 @@ func (bqsa *BigQueryServiceAccount) CreateStorageClient(ctx context.Context) (*s
 	return client, nil
 }
 
+// TableCheck:
+// 1. Creates a table
+// 2. Inserts one row into the table
+// 3. Deletes the table
+func TableCheck(ctx context.Context, client *bigquery.Client, dataset string) error {
+	dummyTable := "peerdb_validate_dummy"
+
+	newTable := client.Dataset(dataset).Table(dummyTable)
+
+	createErr := newTable.Create(ctx, &bigquery.TableMetadata{
+		Schema: []*bigquery.FieldSchema{
+			{
+				Name:     "dummy",
+				Type:     bigquery.BooleanFieldType,
+				Repeated: false,
+			},
+		},
+	})
+	if createErr != nil {
+		return fmt.Errorf("unable to validate table creation within dataset: %v."+
+			"Please check if bigquery.tables.create permission has been granted", createErr)
+	}
+
+	errors := []string{}
```

Comment:
```suggestion
	var errs error[]
```
Using fmt.Errorf with %w instead of %v
Then at end use https://pkg.go.dev/errors#Join
```
return errors.Join(errs...)
```